primitives/xcm-bridge-hub-router/src/lib.rs

```diff
@@ -18,14 +18,25 @@
 
 #![cfg_attr(not(feature = "std"), no_std)]
 
-/// Local XCM channel that may report whether it is congested or not.
-pub trait LocalXcmChannel {
+use xcm::prelude::SendXcm;
+
+/// XCM channel status provider that may report whether it is congested or not.
+pub trait XcmChannelStatusProvider {
 	/// Returns true if the queue is currently congested.
 	fn is_congested() -> bool;
 }
 
-impl LocalXcmChannel for () {
+impl XcmChannelStatusProvider for () {
 	fn is_congested() -> bool {
 		false
 	}
 }
+
+/// XCM channel with dedicated `Sender` and ability to report whether it is congested or not.
+pub trait ToXcmChannel: XcmChannelStatusProvider {
+	type Sender: SendXcm;
+}
+
+impl<T: SendXcm + XcmChannelStatusProvider> ToXcmChannel for T {
```

Comment:
1) IIUC we won't be able to use this blanket implementation, because those queues (HRMP/DMP/UMP) live in different repos (Polkadot/Cumulus) and we are only plugged into Cumulus
2) there are (right now) differences in `is_congested` implementation for the same queue. E.g. at `Source.AH` it just checks if the channel is suspended. At the Target.BH it also checks if there are more than `N` pages in the outbound queue. This is an arguable decision - maybe we don't need to check the number of pages in the outbound queue, but that's how it is implemented now.

Reply:
(that's not the blocker - you may merge as is, but imo blanket impl is not needed here)